cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml

```diff
@@ -107,7 +107,6 @@ spec:
         effect: "NoSchedule"
       - operator: "Exists"
         effect: "NoExecute"
-      #TODO: remove this toleration once #44445 is properly fixed.
```

Comment:
Why remove this comment? The issue hasn't closed.

Reply:
We don't want this toleration to be removed.

Reply:
We don't is different than we never will. TODO indicates we don't but we might want to eventually. Is this issue obsolete? Should we close it?

Reply:
#44445 contains multiple issues.
- The title of the issue: `Improvement: fluentd-gcp to get same toleration as kube-proxy` will be fixed by this PR.
- There was a regression in the daemonset controller mentioned in the bug which was fixed a while back.
- I guess the only thing that is not fixed are the three comments starting from https://github.com/kubernetes/kubernetes/issues/44445#issuecomment-300719212 (users not being able to modify system addons on managed services like GKE and so can't use NoSchedule or NoExecute taints if these addons don't tolerate them because adding such taints make the nodes not have these "required" addons)

But even if we fix that (allow users to modify the toleration of system addons). I think the default should still be that these addons tolerate all taints. If users really want they can use the ability to modify the toleration of system addons to remove these wildcard tolerations.

Also, we need system addons that run on every node to have wildcard NoSchedule toleration for issue #55080, PR #55839.

cc @vishh

Reply:
+1. The expected behavior is that all system addons that are expected to run on all GKE nodes should tolerate all taints and effects. Certain addons like GPU plugins need to tolerate GPU specific taints only.
This I feel is probably slightly different from #44445 where if a user taints all GKE nodes then cluster level system addons will not run at all. This is a separate feature and is not tied to the comment at all.
@mikedanese thoughts?